python/dolfinx/io/gmshio.py

```diff
@@ -161,8 +161,11 @@ def extract_geometry(model: gmsh.model, name: str = None) -> npt.NDArray[np.floa
         assert np.all(indices[perm_sort] == np.arange(len(indices)))
         return points[perm_sort]
 
-    def model_to_mesh(model: gmsh.model, comm: _MPI.Comm, rank: int,
-                      gdim: int = 3) -> typing.Tuple[Mesh, _cpp.mesh.MeshTags_int32, _cpp.mesh.MeshTags_int32]:
+    def model_to_mesh(
+            model: gmsh.model, comm: _MPI.Comm, rank: int,
+            gdim: int = 3,
+            partitioner=create_cell_partitioner(GhostMode.none)) -> typing.Tuple[
```

Comment:
Should we use type hints for partitioner?
something like:
`partitioner: Callable[[Arg1Type, Arg2Type], ReturnType]`